frontend/lib/resolution/call-init-deinit.cpp

```diff
@@ -260,6 +260,12 @@ void CallInitDeinit::checkUseOfDeinited(const AstNode* useAst, ID varId) {
   for (ssize_t i = n - 1; i >= 0; i--) {
     VarFrame* frame = scopeStack[i].get();
     if (frame->deinitedVars.count(varId) > 0) {
+      // For vars dead at end of stmt, don't error for uses within stmt.
+      if (frame->deinitedVars[varId] ==
+          parsing::idToParentId(context, useAst->id())) {
```

Comment:
Yes, it does work in that case and I've added a test to lock it in.

It works because the "ID of stmt after which it is deinited" is actually the ID of the _call_ after which it's deinited. In this case that's the innermost call `baz` which is the one `checkUseOfDeinited` compares against. with `idToParentId`. I'll update comments to accurately reflect what this ID is.